build-and-test.ps1

```diff
@@ -0,0 +1,81 @@
+[cmdletbinding()]
+param(
+    [string]$VersionFilter,
+    [string]$OSFilter,
+    [switch]$IsDryRun
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+$buildFilter = "*"
+if (-not [string]::IsNullOrEmpty($VersionFilter))
+{
+    $buildFilter = "$VersionFilter-$buildFilter"
+}
+if (-not [string]::IsNullOrEmpty($OSFilter))
+{
+    if ([string]::IsNullOrEmpty($VersionFilter))
+    {
+        $buildFilter = "$buildFilter-$OSFilter/*"
+    }
+    else
+    {
+        $buildFilter = "$buildFilter$OSFilter/*"
+    }
+}
+
+function Get-ImageBuilder()
+{
+    $imageBuilderImageName = 'microsoft/dotnet-buildtools-prereqs:image-builder-nanoserver-20180103080524'
+    $imageBuilderContainerName = "ImageBuilder-$(Get-Date -Format yyyyMMddhhmmss)"
+
+    $attempt = 0
+    $maxRetries = 5
+    $waitFactor = 6
+    while ($attempt -lt $maxRetries)
+    {
+        try {
+            New-Item -Path "$imageBuilderFolder" -ItemType Directory -Force | Out-Null
+
+            Invoke-Expression "docker pull $imageBuilderImageName"
+            Invoke-Expression "docker create --name $imageBuilderContainerName $imageBuilderImageName"
+            Invoke-Expression "docker cp ${imageBuilderContainerName}:/image-builder $imageBuilderFolder"
+            Invoke-Expression "docker rmi -f $imageBuilderImageName"
+            break
+        }
+        catch {
+            Write-Warning "$_"
+        }
+
+        $attempt++
+        if ($attempt -ne $maxRetries)
+        {
+            $waitTime = $attempt * $waitFactor
+            Write-Host "Retry ${attempt}/${maxRetries} failed, retrying in $waitTime seconds..."
+            Start-Sleep -Seconds ($waitTime)
+            Remove-Item -Path "$imageBuilderFolder" -Recurse -Force -ErrorAction Continue
+        }
+        else {
+            throw "Retry ${attempt}/${maxRetries} failed, no more retries left."
+        }
+    }
+}
+
+$imageBuilderFolder = [System.IO.Path]::Combine("$PSScriptRoot", ".Microsoft.DotNet.ImageBuilder")
+$imageBuilderExe = [System.IO.Path]::Combine($imageBuilderFolder, "image-builder", "Microsoft.DotNet.ImageBuilder.exe")
```

Comment:
Why create the image-builder sub folder?  This seems to add unnecessary nesting.

Reply:
As discussed offline, this is due to https://github.com/moby/moby/issues/34638